src/pkcs11/core_pkcs11_test.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * FreeRTOS FreeRTOS LTS Qualification Tests preview
+ * Copyright (C) 2022 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file transport_interface_test.h
+ * @brief Test function for corePKCS11 test.
+ */
+#ifndef CORE_PKCS11_TEST_H
+#define CORE_PKCS11_TEST_H
+
+/* Standard includes. */
+#include <stdint.h>
+#include <stdlib.h>
+
+/**
+ * @brief Thread handle data structure definition.
+ */
+typedef void * PkcsTestThreadHandle_t;
+
+/**
+ * @brief Malloc function to allocate memory for pkcs11 test.
+ *
+ * @param[in] size Size in bytes.
+ */
+typedef void * ( * PkcsTestMalloc_t )( size_t size );

```

Comment:
It is a good idea since these functions are shared in multiple test now.
I would like to create a thread_function.h to keep all the thread related interfaces and a memory_function.h to keep memory related functions.
Please help to review these interfaces as well.